### ISSUE TITLE
Bug: no-control-regex does not detect \c control letter escapes

### ISSUE THREAD

**Issue by koreahghg** [eslint/eslint#21281]
**Environment**

Node version: v26.7.0
npm version: 11.19.0
Local ESLint version: 10.9.1
Global ESLint version: N/A
Operating System: macOS 26.4.1

**What parser are you using?**

Default (Espree)

**What did you do?**

`no-control-regex` is documented to disallow regex patterns that explicitly match control characters (U+0000–U+001F), via `\x` hex escapes, `\u` escapes, `\u{}` code point escapes, and raw control characters. It misses one more syntax that does the same thing: the `\c` "control letter" escape (`\cA`–`\cZ` / `\ca`–`\cz`), which is valid under default, `u`, and `v` regex modes and evaluates to the same U+0000–U+001F range.

<details>
<summary>Configuration</summary>

```js
{
  rules: { "no-control-regex": "error" }
}
```
</details>

```js
const knownControl = /\x01/;  // flagged correctly
const missedControl = /\cA/;  // NOT flagged
```

`/\cA/.test(String.fromCharCode(1))` is `true` — `\cA` genuinely matches the control character U+0001, exactly like `\x01` does, but the rule only checks whether the raw source of the matched escape starts with `\x` or `\u` (see `onCharacter` in `lib/rules/no-control-regex.js`), so `\c`-prefixed escapes slip through undetected.

**What did you expect to happen?**

`/\cA/` (and any `\c` + letter A–Z/a–z escape producing a code point in U+0000–U+001F) should be reported the same way `/\x01/` is.

**What actually happened?**

No lint error is produced for `/\cA/`, `new RegExp("\\cA")`, etc., even though they match a control character.

**Link to Minimal Reproducible Example**

https://eslint.org/play/#eyJ0ZXh0IjogImNvbnN0IGtub3duQ29udHJvbCA9IC9cXHgwMS87IC8vIGZsYWdnZWQgY29ycmVjdGx5XG5jb25zdCBtaXNzZWRDb250cm9sID0gL1xcY0EvOyAgLy8gTk9UIGZsYWdnZWQsIGJ1dCAvXFxjQS8udGVzdChTdHJpbmcuZnJvbUNoYXJDb2RlKDEpKSA9PT0gdHJ1ZSIsICJvcHRpb25zIjogeyJydWxlcyI6IHsibm8tY29udHJvbC1yZWdleCI6ICJlcnJvciJ9LCAibGFuZ3VhZ2VPcHRpb25zIjogeyJwYXJzZXJPcHRpb25zIjogeyJlY21hRmVhdHVyZXMiOiB7fX19fX0=

**Participation**

- [x] I am willing to submit a pull request for this issue.

**AI acknowledgment**

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Additional comments**

This bug and a proposed fix were found with the assistance of Claude Code (AI): it read `lib/rules/no-control-regex.js`, identified that the `onCharacter` guard only checks for `\x`/`\u` prefixes, and confirmed the false negative by running `Linter.verify` directly against this repo's code. I reviewed the analysis and reproduction before opening this issue.

I already have a minimal fix + tests + doc update ready locally (extending the `onCharacter` check to also accept `\c`-prefixed escapes) and would like to submit it as a PR once this issue is accepted, per the [AI Usage Policy](https://eslint.org/docs/latest/contribute/ai-policy).

**Comment by DMartens** [comment]
The intent of the rule is to find escapes for control characters which are "hidden" by using hexadecimal escapes.
By using the control letter escape, I think the author knowingly wants to match them (and it is in my opinion a valid fix for the lint message).
This was also discussed [before](https://github.com/eslint/eslint/issues/7476#issuecomment-257124299).
But I think we should mention the control letter escapes in the documentation and add tests for them.
A PR for the documentation and tests is welcome.
